modules/programs/ledger.nix

```diff
@@ -0,0 +1,196 @@
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Specify the input date format for journal entries.
+      '';
+    };
+    masterAccount = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Prepend all account names with the argument.
+      '';
+    };
+    noAliases = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Ledger does not expand any aliases if this option is specified.
+      '';
+    };
+    pedantic = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Accounts, tags or commodities not previously declared will cause errors.
+      '';
+    };
+    permissive = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Quiet balance assertions.
+      '';
+    };
+    priceDB = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Specify the location of the price entry data file.
+      '';
+    };
+    priceExpectedFreshness = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      description = ''
+        Set the expected freshness of price quotes, in minutes. That
+        is, if the last known quote for any commodity is older than
+        this value, and if --download is being used, then the Internet
+        will be consulted again for a newer price. Otherwise, the old
+        price is still considered to be fresh enough.
+      '';
+    };
+    strict = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Ledger normally silently accepts any account or commodity in a
+        posting, even if you have misspelled a commonly used one. The
+        option --strict changes that behavior. While running with
+        --strict, Ledger interprets all cleared transactions as
+        correct, and if it encounters a new account or commodity (same
+        as a misspelled commodity or account) it will issue a warning
+        giving you the file and line number of the problem.
+      '';
+    };
+    recursiveAliases = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Normally, ledger only expands aliases once. With this option,
+        ledger tries to expand the result of alias expansion
+        recursively, until no more expansions apply.
+      '';
+    };
+    timeColon = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        The --time-colon option will display the value for a seconds
+        based commodity as real hours and minutes.
+      '';
+    };
+    valueExpr = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Set a global value expression annotation.
+      '';
+    };
+    extraConfig = mkOption {
+      type = types.lines;
```

Comment:
It is [recommended to use an attribute set](https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md#part-1-structural-settings-instead-of-stringly-extraconfig) named `settings` instead of a string.